packages/cdk/lib/stacks/engines/snakemake-engine-construct.ts

```diff
@@ -0,0 +1,154 @@
+import { Aws, Stack } from "aws-cdk-lib";
+import { SnakemakeEngine } from "../../constructs/engines/snakemake/snakemake-engine";
+import { EngineOptions } from "../../types";
+import { ApiProxy, Batch } from "../../constructs";
+import { EngineOutputs, EngineConstruct } from "./engine-construct";
+import { ILogGroup } from "aws-cdk-lib/aws-logs";
+import { ComputeResourceType } from "@aws-cdk/aws-batch-alpha";
+import { ENGINE_SNAKEMAKE } from "../../constants";
+import { Construct } from "constructs";
+import { IRole, ManagedPolicy, PolicyDocument, PolicyStatement, Role, ServicePrincipal } from "aws-cdk-lib/aws-iam";
+import { IVpc } from "aws-cdk-lib/aws-ec2";
+import { ContextAppParameters } from "../../env";
+import { HeadJobBatchPolicy } from "../../roles/policies/head-job-batch-policy";
+import { BatchPolicies } from "../../roles/policies/batch-policies";
+import { Bucket, IBucket } from "aws-cdk-lib/aws-s3";
+import { BucketOperations } from "../../common/BucketOperations";
+import { LaunchTemplateData } from "../../constructs/launch-template-data";
+
+export class SnakemakeEngineConstruct extends EngineConstruct {
+  public readonly apiProxy: ApiProxy;
+  public readonly adapterLogGroup: ILogGroup;
+  public readonly snakemakeEngine: SnakemakeEngine;
+  private readonly batchHead: Batch;
+  private readonly batchWorkers: Batch;
+  private readonly outputBucket: IBucket;
+
+  constructor(scope: Construct, id: string, props: EngineOptions) {
+    super(scope, id);
+
+    const { vpc, contextParameters } = props;
+    const params = props.contextParameters;
+
+    this.batchHead = this.renderBatch("HeadBatch", vpc, contextParameters, ComputeResourceType.FARGATE);
+    const workerComputeType = contextParameters.requestSpotInstances ? ComputeResourceType.SPOT : ComputeResourceType.ON_DEMAND;
+    this.batchWorkers = this.renderBatch("TaskBatch", vpc, contextParameters, workerComputeType);
+
+    this.snakemakeEngine = new SnakemakeEngine(this, "SnakemakeEngine", {
+      vpc: props.vpc,
+      engineBatch: this.batchHead,
+      workerBatch: this.batchWorkers,
+      rootDirS3Uri: params.getEngineBucketPath(),
+    });
+
+    this.batchHead.role.addToPrincipalPolicy(
+      new PolicyStatement({
+        actions: ["elasticfilesystem:DescribeAccessPoints"],
+        resources: [this.snakemakeEngine.fileSystem.fileSystemArn, this.snakemakeEngine.fsap.accessPointArn],
+      })
+    );
+    this.batchHead.role.attachInlinePolicy(new HeadJobBatchPolicy(this, "HeadJobBatchPolicy"));
+    this.batchHead.role.addToPrincipalPolicy(
+      new PolicyStatement({
+        actions: ["batch:TagResource"],
+        resources: ["*"],
```

Comment:
This should be limited to the child jobs

Reply:
Since snakemake submits the jobs we can't know the arns of the jobs we want to tag hence this is necessary. Same holds true for the tag:GetResources comment below.